fabric-resource-loader-v0/src/main/java/net/fabricmc/fabric/api/resource/ResourceManagerHelper.java

```diff
@@ -49,6 +52,16 @@ default void addReloadListener(IdentifiableResourceReloadListener listener) {
 	 */
 	void registerReloadListener(IdentifiableResourceReloadListener listener);
 
+	/**
+	 * Register a resource reload listener for a given resource manager type.
+	 *
+	 * <p>Note: This is only supported for server data reload listeners.
+	 *
+	 * @param identifier The identifier of the listener.
+	 * @param listenerFactory   A function that creates a new instance of the listener with a given registry lookup.
+	 */
+	void registerReloadListener(Identifier identifier, Function<RegistryWrapper.WrapperLookup, IdentifiableResourceReloadListener> listenerFactory);
```

Comment:
I know that this is already merged, but what is the point of having an Identifier here? We're already registering an **Identifiable**ResourceReloadListener and later this just gets wrapped.

Wouldn't this be better something like this?
```java
void registerReloadListener(Function<RegistryWrapper.WrapperLookup, IdentifiableResourceReloadListener> listenerFactory);

void registerReloadListener(Function<RegistryWrapper.WrapperLookup, Collection<IdentifiableResourceReloadListener>> listenersFactory);
```

Since the helper does not expose all reloaders to the API, we could change the internals to support something like this?

Reply:
We need the ID before creating the listner, this was very much a stop gap solution that didnt require re-writing the whole module. `IdentifiableResourceReloadListener` is a very old API, and is not how we would create this kind of API these days.

At somepoint I hope there will be a replacement resource loader module that improves this.